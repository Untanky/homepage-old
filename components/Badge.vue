<template>
  <span
    class="badge"
    :class="type"
  >
    <slot />
  </span>
</template>

<script>
const allowedTypes = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  BLACK: 'black',
  WHITE: 'white',
};

export default {
  name: 'Badge',
  props: {
    type: {
      type: String,
      required: false,
      default: () => 'primary',
      validator: (value) => Object.values(allowedTypes).includes(value),
    },
  },
};
</script>

<style lang="scss" scoped>
@import "../assets/scss/variables";

.badge {
  display: inline-block;
  width: auto;
  padding: 0.125em 1em;
  border-radius: $border-radius;
  box-shadow: $shadow;

  &.primary {
    background-color: $primary;
    color: $text-on-primary;
  }

  &.secondary {
    background-color: $secondary;
    color: $text-on-secondary;
  }

  &.black {
    background-color: black;
    color: white;
  }

  &.white {
    background-color: white;
    color: black;
  }
}
</style>
