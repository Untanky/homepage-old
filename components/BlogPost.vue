<template>
  <div class="blog-post">
    <header>
      <h1 class="title">
        {{ article.title }}
      </h1>
      <div class="meta-data">
        <div class="author">
          by {{ article.author }}
        </div>
        <div class="create date">
          Created at {{ createdDate }}
        </div>
        <div class="update date">
          Last updated at {{ updatedDate }}
        </div>
      </div>
    </header>
    <article>
      <post-section
        v-for="(section, index) in article.content"
        :key="index"
        :layout="section.layout"
        :type="section.type"
        :content="section.content"
      />
    </article>
  </div>
</template>

<script>
import { DateTime } from 'luxon';

import PostSection from './PostSection.vue';

import article from '../article.test.json';

// eslint-disable-next-line no-unused-vars
export const MediaTypes = {
  SECTION: 'TITLE',
  TEXT: 'TEXT',
  VIDEO: 'YOUTUBE_VIDEO',
  IMAGE: 'IMAGE',
  CODE: 'CODE',
  LINK: 'LINK',
};

export default {
  name: 'BlogPost',
  components: {
    PostSection,
  },
  data: () => ({
    article,
  }),
  computed: {
    createdDate() {
      return DateTime.fromISO(this.article.created).toLocaleString(DateTime.DATETIME_MED);
    },
    updatedDate() {
      return DateTime.fromISO(this.article.updated).toLocaleString(DateTime.DATETIME_MED);
    },
  },
};
</script>

<style lang="scss" scoped>
@import "../assets/scss/variables";
@import "../assets/scss/responsive";

$single-width: 600px;
$double-width: 900px;

.blog-post {
  background-color: white;
  max-width: 1000px;
  border-radius: $border-radius;
  padding: 2em 1em 0.25em;
  margin: 0 auto;
  margin-bottom: 72px;
  box-shadow: $shadow;

  @include tablet-up {
    margin-top: 64px;
    margin-bottom: 0;
  }

  header {
    .title {
      font-size: 44px;
      text-align: center;

      @include tablet-up {
        margin: 1em 0;
      }

      @include laptop-up {
        margin: 2em 0;
      }
    }

    .meta-data {
      padding: 0.5em 0;
      border-bottom: $border-primary;
      margin: 0 auto;

      @include tablet-up {
        max-width: $single-width;
      }

      .author {
        font-size: 20px;
        text-align: left;
      }

      .date {
        font-size: 12px;
        text-align: left;
      }
    }
  }

  article {
    text-align: justify;
  }
}
</style>
