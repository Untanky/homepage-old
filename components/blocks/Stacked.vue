<template>
  <div
    class="stacked flex h-full"
    :class="[flexDirection, data.classes]"
  >
    <component
      v-for="(componentData,index) in data.componentData"
      :key="index"
      :is="componentData.setting.component"
      :data="componentData.data"
    />
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';
import Stacked, { StackDirection } from '../../src/components/Stacked';

export default Vue.extend({
  name: 'Stacked',
  props: {
    data: {
      type: Object as PropType<Stacked>,
      required: true,
    },
  },
  computed: {
    flexDirection() {
      switch (this.data.direction) {
        case StackDirection.Vertical:
          return 'flex-col';
        case StackDirection.Horizontal:
          return 'flex-row';
        default:
          return '';
      }
    }
  }
});
</script>
