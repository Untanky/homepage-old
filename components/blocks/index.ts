import formattedText from './FormattedText.vue';
import fullImage from './FullImage.vue';
import keyValue from './KeyValue.vue';
import richLink from './RichLink.vue';
import stacked from './Stacked.vue';
import timeline from './Timeline.vue';

export const FormattedText = formattedText;
export const FullImage = fullImage;
export const KeyValue = keyValue;
export const RichLink = richLink;
export const Stacked = stacked;
export const Timeline = timeline;
