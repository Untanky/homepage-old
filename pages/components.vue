<template lang="pug">
  div
    rich-link(linkUrl="https://www.google.com" title="GitHub" text="Untanky" imageUrl="https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png")
    timeline(:elements="timelineElements")
      template(v-for="(element, i) in timelineElements.list" v-slot:[`timeline-event__${i}`])
        stacked-experience-text(:key="i" :element="element.content")
    card(title="Test" linkText="Test")
      div Test text that is very long and respective of normal content
      div Another text that is equally meaningful as the first block of content
    card( linkText="Test")
      div Test text that is very long and respective of normal content
      div Another text that is equally meaningful as the first block of content
    card( linkText="Test")
      div Test text that is very long and respective of normal content
      div Another text that is equally meaningful as the first block of content
    card(title = "Test")
      div Test text that is very long and respective of normal content
      div Another text that is equally meaningful as the first block of content
</template>

<script lang="ts">
import Vue from 'vue';
import { DateTime } from 'luxon';
import RichLink from '../components/RichLink.vue';
import Timeline from '../components/Timeline.vue';
import Card from '../components/Card.vue';
import StackedExperienceText from '../components/StackedExperienceText.vue';

export default Vue.extend({
  components: {
    RichLink,
    Timeline,
    Card,
    StackedExperienceText,
  },
  data: () => ({
    timelineElements: {
      list: [
        {
          content: {
            title: 'Pinuts media+science Multimedia-Agentur',
            subtitle: 'Web Developer',
            description: 'Development of different web application in a legacy web service framework. Frontend Development with JSX.',
          },
          timelineEntry: {
            from: DateTime.fromObject({ year: 2020, month: 8 }),
            to: undefined,
            active: true,
          },
        },
        {
          content: {
            title: 'WBS Coding School',
            subtitle: 'Software Developer',
            description: 'Full-Stack Development of web applications with MERN-stack related to a teaching platform and other business needs.',
          },
          timelineEntry: {
            from: DateTime.fromObject({ year: 2020, month: 2 }),
            to: DateTime.fromObject({ year: 2020, month: 5 }),
            active: false,
          },
        },
        {
          content: {
            title: 'Hochschule f√ºr Technik und Wirtschaft',
            subtitle: 'Student Assistant for the course "theorectic computer science"',
            description: 'Creating homework exercises for students and grading them. Giving tutorial lessons',
          },
          timelineEntry: {
            from: DateTime.fromObject({ year: 2018, month: 10 }),
            to: DateTime.fromObject({ year: 2020, month: 7 }),
            active: false,
          },
        },
      ],
    },
  }),
});
</script>
