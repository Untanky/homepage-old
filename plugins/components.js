import Vue from 'vue';
import FullImage from '../components/FullImage.vue';
import StackedText from '../components/StackedText.vue';

Vue.component('full-image', FullImage);
Vue.component('stacked-text', StackedText);
