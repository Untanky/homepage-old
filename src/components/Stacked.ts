/* eslint-disable no-unused-vars */
import ComponentData from '../interfaces/ComponentData';
import LayoutComposition from '../interfaces/LayoutComposition';

export enum StackDirection {
  Vertical = 'VERTICAL',
  Horizontal = 'HORIZONTAL',
}

interface Stacked extends ComponentData {
  direction: StackDirection;
  componentData: LayoutComposition<any>[];
  classes: string[];
}

export default Stacked;
