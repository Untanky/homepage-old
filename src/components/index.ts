export { default as FormattedText } from './FormattedText';
export { default as FullImage } from './FullImage';
export { default as KeyValue } from './KeyValue';
export { default as RichLink } from './RichLink';
export { default as Stacked } from './Stacked';
export { default as TimelineElement } from './TimelineElement';
export { default as TimelineEntry } from './TimelineEntry';
export { default as TimelineList } from './TimelineList';
