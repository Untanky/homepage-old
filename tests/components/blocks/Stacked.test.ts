/* eslint-disable no-undef */
// global it, expect
import { mount } from '@vue/test-utils';
import { StackDirection } from '../../../src/components/Stacked';
import Stacked from '../../../components/blocks/Stacked.vue';

describe('Stacked', () => {
  it('should run with vertical stack', () => {
    const props = {
      data: {
        direction: StackDirection.Vertical,
        componentData: [
          {
            setting: {
              component: 'div',
            },
            data: {
              hello: 'world',
            },
          },
          {
            setting: {
              component: 'div',
            },
            data: {
              goodbye: 'world',
            },
          },
        ],
        classes: [],
      },
    };

    const comp = mount(Stacked, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with horizontal stack', () => {
    const props = {
      data: {
        direction: StackDirection.Horizontal,
        componentData: [
          {
            setting: {
              component: 'div',
            },
            data: {
              hello: 'world',
            },
          },
          {
            setting: {
              component: 'div',
            },
            data: {
              goodbye: 'world',
            },
          },
        ],
        classes: [],
      },
    };

    const comp = mount(Stacked, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run without direction', () => {
    const props = {
      data: {
        direction: '',
        componentData: [
          {
            setting: {
              component: 'div',
            },
            data: {
              hello: 'world',
            },
          },
          {
            setting: {
              component: 'div',
            },
            data: {
              goodbye: 'world',
            },
          },
        ],
        classes: [],
      },
    };

    const comp = mount(Stacked, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run without data', () => {
    const props = {
      data: {
        direction: StackDirection.Horizontal,
        componentData: [],
        classes: [],
      },
    };

    const comp = mount(Stacked, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });

  it('should run with classes', () => {
    const props = {
      data: {
        direction: StackDirection.Horizontal,
        componentData: [
          {
            setting: {
              component: 'div',
            },
            data: {
              hello: 'world',
            },
          },
          {
            setting: {
              component: 'div',
            },
            data: {
              goodbye: 'world',
            },
          },
        ],
        classes: ['test-class', 'my-class'],
      },
    };

    const comp = mount(Stacked, { propsData: props });
    expect(comp.element).toMatchSnapshot();
  });
});
