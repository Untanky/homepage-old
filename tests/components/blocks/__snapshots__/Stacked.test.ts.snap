// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stacked should run with classes 1`] = `
<div
  class="stacked flex flex-col test-class my-class"
>
  <div
    data="[object Object]"
  />
  <div
    data="[object Object]"
  />
</div>
`;

exports[`Stacked should run with horizontal stack 1`] = `
<div
  class="stacked flex flex-col"
>
  <div
    data="[object Object]"
  />
  <div
    data="[object Object]"
  />
</div>
`;

exports[`Stacked should run with vertical stack 1`] = `
<div
  class="stacked flex flex-row"
>
  <div
    data="[object Object]"
  />
  <div
    data="[object Object]"
  />
</div>
`;

exports[`Stacked should run without data 1`] = `
<div
  class="stacked flex flex-col"
/>
`;

exports[`Stacked should run without direction 1`] = `
<div
  class="stacked flex"
>
  <div
    data="[object Object]"
  />
  <div
    data="[object Object]"
  />
</div>
`;
